assignments/week_1/exercise_1_2/exercise_1_2.md

```diff
@@ -0,0 +1,231 @@
+# Exercise 1.2
+Welcome to your second exercise of RoboJackets software training! This exercise will be testing your knowledge of basic C++ syntax and vector manipulation. For this exercise we expect you to have watched or know the content of the Codecademy C++ course up through vectors.
+
+# Exercise Objective
+In this exercise, you will be reversing an vector and checking if a string is a palindrome. The purpose of this exercise is to get you familar with working with vectors and strings. This exercise should help solidify your knowledge of loops, variables, conditionals, and vectors. _We aren't going to be using any new C++ standard library functions in the instructions, but feel free to use them if you want._
+
+# Starting Code
+We have already given you a main() function to write your code in and have included the proper libraries for you to use. We have also set up most of variable initilization and commented where you will write your code.
+
+
+### Part 1: Vector Reversal
+```c++
+#include <iostream>
+#include <vector>
+
+int main()
+{
+    // Part 1: Vector Reversal
+
+    // Input vector from command-line
+    std::vector<int> list;
+    int input;
+    std::cout << "Input List Of Numbers (end list with non-integer):" << std::endl;
+    while (std::cin >> input)
+        list.push_back(input);
+
+    // Print original vector
+    std::cout << "Original Vector: ";
+    for (int v : list)
+        std::cout << v << " ";
+    std::cout << std::endl;
+
+    // Reverse Vector
+    // WRITE YOUR VECTOR REVERSAL HERE
+
+    // Print the Reversed vector
+    std::cout << "Reversed Vector: ";
+    for (int v : list)
+        std::cout << v << " ";
+    std::cout << std::endl;
+
+    return 0;
+}
+```
+
+1. Create variables for the vector's `start` and `end` indices
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    int start = 0;
+    int end = list.size() - 1;
+  ```
+
+</details>
+
+2. Create a loop which continues as long as `start` is less than `end`
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    while (start < end)
+    {
+        // MORE CODE
+    }
+  ```
+
+</details>
+
+3. In the loop, increment `start` and decrement `end` so that `start` will iterate forwards over the first half and `end` will iterate backwards over the second half
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    while (start < end)
+    {
+        // MORE CODE
+
+        start++;
+        end--;
+    }
+  ```
+
+</details>
+
+4. In the loop, swap the elements at `start` and `end` with each other so `list[start]` -> `list[end]` and `list[end]` -> `list[start]`
+<details>
+  <summary>Answer</summary>
+
+  ```c++
+    while (start < end)
+    {
+        int temp = list[start];
+        list[start] = list[end];
+        list[end] = temp;
+        start++;
+        end--;
+    }
+  ```
+
+</details>
+
+Now try to run the program and test that the vector gets reversed!
+
+
```

Comment:
Add an introduction for this part of the exercise. Give a brief explanation of what a palindrome is and example input/output for this part.